Framework/CurveFitting/inc/MantidCurveFitting/ComplexMatrix.h

```diff
@@ -0,0 +1,461 @@
+      : m_1(m1.matrix), m_2(mm.m_1), m_3(mm.m_2), tr1(true), tr2(mm.tr1),
+        tr3(mm.tr2), ctr1(false), ctr2(mm.ctr1), ctr3(mm.ctr2) {}
+
+  ComplexMatrixMult3(const ComplexMatrixMult2 &mm, const ComplexMatrix &m2)
+      : m_1(mm.m_1), m_2(mm.m_2), m_3(m2), tr1(mm.tr1), tr2(mm.tr2), tr3(false),
+        ctr1(mm.ctr1), ctr2(mm.ctr2), ctr3(false) {}
+
+  ComplexMatrixMult3(const ComplexMatrixMult2 &mm, const Tr<ComplexMatrix> &m2)
+      : m_1(mm.m_1), m_2(mm.m_2), m_3(m2.matrix), tr1(mm.tr1), tr2(mm.tr2),
+        tr3(true), ctr1(mm.ctr1), ctr2(mm.ctr2), ctr3(false) {}
+
+  ComplexMatrixMult3(const CTr &m1, const ComplexMatrixMult2 &mm)
+      : m_1(m1.matrix), m_2(mm.m_1), m_3(mm.m_2), tr1(false), tr2(mm.tr1),
+        tr3(mm.tr2), ctr1(true), ctr2(mm.ctr1), ctr3(mm.ctr2) {}
+
+  ComplexMatrixMult3(const ComplexMatrixMult2 &mm, const CTr &m2)
+      : m_1(mm.m_1), m_2(mm.m_2), m_3(m2.matrix), tr1(mm.tr1), tr2(mm.tr2),
+        tr3(false), ctr1(mm.ctr1), ctr2(mm.ctr2), ctr3(true) {}
+};
+
+/**
+A complex-valued matrix for linear algebra computations.
+The '*' operator is overloaded to help with
+matrix multiplication.
+
+Copyright &copy; 2010 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+National Laboratory & European Spallation Source
+
+This file is part of Mantid.
+
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>
+Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+class MANTID_CURVEFITTING_DLL ComplexMatrix {
+public:
+  /// Constructor
+  ComplexMatrix();
+  /// Constructor
+  ComplexMatrix(const size_t nx, const size_t ny);
+  /// Copy constructor
+  ComplexMatrix(const ComplexMatrix &M);
+  /// Create a submatrix.
+  ComplexMatrix(const ComplexMatrix &M, size_t row, size_t col, size_t nRows,
+                size_t nCols);
+  /// Create this matrix from a product of two other matrices
+  ComplexMatrix(const ComplexMatrixMult2 &mult2);
+  /// Create this matrix from a product of three other matrices
+  ComplexMatrix(const ComplexMatrixMult3 &mult3);
+  /// Destructor.
+  ~ComplexMatrix();
+
+  /// Copy assignment operator
+  ComplexMatrix &operator=(const ComplexMatrix &M);
+  /// Copy assignment operator
+  ComplexMatrix &operator=(const gsl_matrix_complex *M);
+
+  /// Get the pointer to the GSL matrix
+  gsl_matrix_complex *gsl() { return m_matrix; }
```

Comment:
It is a GSL interface of this class. Any operation you do on gsl_matrix_complex will be reflected in ComplexMatrix.